claims.go

```diff
@@ -89,7 +90,13 @@ func validateAudience(boundAudiences, audClaim []string, strict bool) error {
 
 // validateBoundClaims checks that all of the claim:value requirements in boundClaims are
 // met in allClaims.
-func validateBoundClaims(logger log.Logger, boundClaims, allClaims map[string]interface{}) error {
+func validateBoundClaims(logger log.Logger, boundClaimsType string, boundClaims, allClaims map[string]interface{}) error {
+	if boundClaimsType != boundClaimsTypeString && boundClaimsType != boundClaimsTypeGlob {
```

Comment:
Since we're checking for a valid type when the role is created, I think checking again here in an internal function is probably unnecessary.

Reply:
Removed the check in the new commit (to simply reviewing it)